packages/lib/checkRateLimitAndThrowError.ts

```diff
@@ -27,6 +31,75 @@ export async function checkRateLimitAndThrowError({
   return response;
 }
 
+async function checkRateLimitForNextJs({
+  rateLimitingType = "core",
+  identifier,
+  opts,
+}: Pick<RateLimitHelper, "rateLimitingType" | "identifier" | "opts">): Promise<
+  { success: true } | { success: false; error: { statusCode: number; message: string } }
+> {
+  try {
+    await checkRateLimitAndThrowError({ rateLimitingType, identifier, opts });
+    return { success: true };
+  } catch (error) {
+    if (error instanceof HttpError) {
+      return {
+        success: false,
+        error: {
+          statusCode: error.statusCode,
+          message: error.message,
+        },
+      };
+    }
+    throw error;
+  }
+}
+
+/**
+ * Sets context.res.statusCode to 429 and returns props with errorMessage when rate limited.
+ * This ensures the 429 status code is properly returned (unlike notFound which returns 404).
+ *
+ * @param context - Next.js GetServerSidePropsContext
+ * @param identifierSuffix - Suffix to append to the rate limit identifier (e.g., "[user]/[type]")
+ * @param rateLimitingType - Type of rate limiting (default: "core") (see RateLimitHelper["rateLimitingType"])
+ * @param opts - Optional rate limit options
+ * @returns null if rate limit passes, or a Next.js response object with props containing errorMessage if rate limited
+ *
+ * @example
+ * ```ts
+ * export const getServerSideProps = async (context: GetServerSidePropsContext) => {
+ *   const rateLimitResponse = await handleRateLimitForNextJs(context, "[user]/[type]");
+ *   if (rateLimitResponse) return rateLimitResponse;
+ *
+ *   // Continue with normal logic...
+ * }
+ * ```
+ */
+export async function handleRateLimitForNextJs(
+  context: GetServerSidePropsContext,
+  identifierSuffix: string,
+  rateLimitingType: RateLimitHelper["rateLimitingType"] = "core",
+  opts?: RateLimitHelper["opts"]
+): Promise<{ props: { errorMessage: string } } | null> {
+  const requestorIp = getIP(context.req as unknown as Request);
+  const rateLimitResult = await checkRateLimitForNextJs({
+    rateLimitingType,
+    identifier: `${identifierSuffix}-${piiHasher.hash(requestorIp)}`,
+    opts,
+  });
+
```

Comment:
We should make a more generic function here. For example : if we want to reuse user id instead of user ip for ratelimiting then we would have to duplicate the whole pattern